File: backend/src/service/timeline.service.ts
import Article from "@/models/Article"

export async function getTimelineList(page: number, size: number) {
    const skip = (page - 1) * size
    const filter = { deleted: { $ne: true }, status: 'PUBLIC' }

    const articles = await Article.find(filter)
        .sort({ createdAt: -1 })
        .select('_id title published cover')  // 按时间倒序（最新的在前）
        .skip(skip)
        .limit(size)
        .lean()

    return articles
}

File: backend/java-backend/src/main/java/com/example/demo/service/TimelineService.java
package com.example.demo.service;

import com.example.demo.model.mongo.Article;
import com.example.demo.repository.mongo.ArticleRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;
import org.springframework.data.domain.Sort;
import org.springframework.stereotype.Service;

import java.util.*;

@Service
public class TimelineService {

    @Autowired
    private ArticleRepository articleRepository;

    public Map<String, Object> getTimelineList(int page, int size) {
        PageRequest pageRequest = PageRequest.of(page - 1, size, Sort.by(Sort.Direction.DESC, "createdAt"));
        Page<Article> articlePage = articleRepository.findByDeletedNotAndStatus(true, "PUBLIC", pageRequest);

        List<Map<String, Object>> articleList = new ArrayList<>();
        for (Article article : articlePage.getContent()) {
            Map<String, Object> item = new LinkedHashMap<>();
            item.put("id", article.getId());
            item.put("title", article.getTitle());
            item.put("published", article.getPublished());
            item.put("cover", article.getCover());
            articleList.add(item);
        }

        Map<String, Object> result = new HashMap<>();
        result.put("list", articleList);
        result.put("total", articlePage.getTotalElements());
        result.put("page", page);
        result.put("size", size);
        return result;
    }
}
